src/main/java/apoc/custom/CypherProcedures.java

```diff
@@ -364,30 +370,76 @@ public static GraphPropertiesProxy getProperties(GraphDatabaseAPI api) {
         }
 
         private void restoreProcedures() {
+            timer = new Timer(this.getClass().getSimpleName());
             CustomStatementRegistry registry = new CustomStatementRegistry(api, log);
-            Map<String, Map<String,Map<String, Object>>> stored = readData(properties);
-            stored.get(FUNCTIONS).forEach((name, data) -> {
-                registry.registerFunction(name, (String) data.get("statement"), (String) data.get("output"),
-                        (List<List<String>>) data.get("inputs"), (Boolean)data.get("forceSingle"), (String) data.get("description"));
-            });
-            stored.get(PROCEDURES).forEach((name, data) -> {
-                registry.registerProcedure(name, (String) data.get("statement"), (String) data.get("mode"),
-                        (List<List<String>>) data.get("outputs"), (List<List<String>>) data.get("inputs"), (String) data.get("description"));
-            });
+            timer.scheduleAtFixedRate(new TimerTask() {
+                private AtomicLong highWaterMark = new AtomicLong(0);
+                private static final String LAST_UPDATE_KEY = "lastUpdate";
+                @Override
+                public void run() {
+                    Map<String, Map<String, Map<String, Object>>> storage = readData(properties);
```

Comment:
can we extract this into a number of methods to make the code more readable

Reply:
also can we keep the simple approach from before where we have one method for functions and one for procedures.